docs/core/whats-new/dotnet-10/libraries.md

```diff
@@ -0,0 +1,267 @@
+|-----------|----------|-----------|---------------|----------:|----------:|-----------:|----------:|------:|--------:|--------:|--------:|----------:|------------:|
+| Benchmark | Baseline | .NET 9.0  | 2             | 2.729 μs  | 0.0538 μs | 0.0449 μs  | 2.706 μs  | 1.00  | 0.02    | 2.2697  | -       | 9.28 KB   | 1.00        |
+| Benchmark | CoreRun  | .NET 10.0 | 2             | 1.665 μs  | 0.0256 μs | 0.0239 μs  | 1.659 μs  | 0.61  | 0.01    | 2.2259  | -       | 9.1 KB    | 0.98        |
+|           |          |           |               |           |           |            |           |       |         |         |         |           |             |
+| Benchmark | Baseline | .NET 9.0  | 10            | 10.341 μs | 0.1988 μs | 0.2289 μs  | 10.266 μs | 1.00  | 0.03    | 9.7046  | -       | 39.76 KB  | 1.00        |
+| Benchmark | CoreRun  | .NET 10.0 | 10            | 7.937 μs  | 0.1514 μs | 0.2487 μs  | 7.831 μs  | 0.77  | 0.03    | 9.5215  | -       | 39.02 KB  | 0.98        |
+|           |          |           |               |           |           |            |           |       |         |         |         |           |             |
+| Benchmark | Baseline | .NET 9.0  | 25            | 24.677 μs | 0.4903 μs | 0.8842 μs  | 24.563 μs | 1.00  | 0.05    | 20.1721 | 3.3569  | 82.92 KB  | 1.00        |
+| Benchmark | CoreRun  | .NET 10.0 | 25            | 18.247 μs | 0.3474 μs | 0.3412 μs  | 18.192 μs | 0.74  | 0.03    | 19.7754 | 3.2654  | 81.13 KB  | 0.98        |
+|           |          |           |               |           |           |            |           |       |         |         |         |           |             |
+| Benchmark | Baseline | .NET 9.0  | 50            | 67.420 μs | 5.7447 μs | 16.9384 μs | 57.185 μs | 1.05  | 0.35    | 40.5273 | 13.4888 | 166.71 KB | 1.00        |
+| Benchmark | CoreRun  | .NET 10.0 | 50            | 41.443 μs | 0.7212 μs | 0.8306 μs  | 41.493 μs | 0.65  | 0.13    | 39.6729 | 0.0610  | 163.16 KB | 0.98        |
+
+Additional benchmarking details provided [in the PR description](https://github.com/dotnet/runtime/pull/103153#issue-2339713028).
+
+## Additional `TryAdd` and `TryGetValue` overloads for `OrderedDictionary<TKey, TValue>`
+
+`OrderedDictionary<TKey, TValue>` provides `TryAdd` and `TryGetValue` for addition and retrieval like any other `IDictionary<TKey, TValue>` implementation. However, there are scenarios where you might want to perform additional operations, so new overloads have been added which return an index to the entry:
+
+```cs
+public class OrderedDictionary<TKey, TValue>
+{
+    // New overloads
+    public bool TryAdd(TKey key, TValue value, out int index);
+    public bool TryGetValue(TKey key, out TValue value, out int index);
+}
+```
+
+This index can then be used with `GetAt`/`SetAt` for fast access to the entry. An example usage of the new `TryAdd` overload is to add or update a key/value pair in the ordered dictionary:
```

Comment:
Add xrefs for GetAt/SetAt. Do you know about the tool David, Andy, and I wrote to help with xrefs?